### PR TITLE
(RE-5565) Resolve PerceivedComplexity rubocop violations for more maintainable code

### PR DESCRIPTION
**Pull request by haus**
This PR refactors 4 problematic methods that were triggering PerceivedComplexity[1] cops. The methods in question were deeply nested with a lot of conditional logic. The refactors took a few paths to alleviate this: removing unneeded conditionals and refactoring behavior into smaller methods. Some of the refactoring included adding new subclasses for AIX and SLES. The final commit reenables the PerceivedComplexity cop in vanagon.

[1] - http://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Metrics/PerceivedComplexity

### CODE REVIEW

**Comment by haus** [comment]
http://jenkins-staging.delivery.puppetlabs.net/job/release_vanagon_intn-sys_master/68/ green build against changes.

File: lib/vanagon/engine/pooler.rb [hunk 1]
```diff
@@ -37,7 +37,7 @@ def select_target
           '{"' + @platform.vmpooler_template + '":"1"}',
           { 'X-AUTH-TOKEN' => @token }
         )
-        if response and response["ok"]
+        if response["ok"]
```
Comment (stahnma) [inline lib/vanagon/engine/pooler.rb]: Will this give nil::nillclass errors if response isn't there? Can that happen?

Comment (haus) [inline lib/vanagon/engine/pooler.rb]: http_request could return an empty hash, but will never return nil. http_request would raise an error in the nil case. indexing into a non-existent key on a hash returns nil.

File: lib/vanagon/platform/dsl.rb [hunk 2]
```diff
@@ -198,83 +205,25 @@ def codename(name)
       #
       # @param definition [String] the repo setup file, must be a valid uri, fetched with curl
       # @param gpg_key [String] optional gpg key to be fetched via curl and installed
+      # @deprecated Please use the add_build_repository DSL method instead. apt_repo will be removed in a future vanagon release.
       def apt_repo(definition, gpg_key = nil)
-        # i.e., definition = http://builds.delivery.puppetlabs.net/puppet-agent/0.2.1/repo_configs/deb/pl-puppet-agent-0.2.1-wheezy.list
```
Comment (haus) [inline lib/vanagon/platform/dsl.rb]: @haus add some warnings here.

Comment (stahnma) [inline lib/vanagon/platform/dsl.rb]: As discussed we should warn for deprecation.

**Comment by haus** [comment]
@stahnma warnings added.

**Comment by stahnma** [comment]
:+1:

**Comment by stahnma** [comment]
Did we try AIX with it yet?

**Comment by haus** [comment]
nope

File: lib/vanagon/platform/rpm.rb [hunk 3]
```diff
@@ -48,10 +48,29 @@ def rpm_defines
...
+
+      def add_repository(definition)
+        definition = URI.parse(definition)
+
+        commands = ["rpm -q curl > /dev/null || yum -y install curl"]
+        if definition.scheme =~ /^(http|ftp)/
+          if File.extname(definition.path) == '.rpm'
+            # repo definition is an rpm (like puppetlabs-release)
+            commands << "curl -o local.rpm '#{definition}'; rpm -Uvh local.rpm; rm -f local.rpm"
+          else
```
Comment (melissa) [inline lib/vanagon/platform/rpm.rb]: should we also be installing a gpg key in this case, like we are for deb? It's probably not something for this PR, but something to think about

**Comment by melissa** [comment]
the logic seems solid to me.

I'm :+1: as long as it blends

File: lib/vanagon/platform/rpm.rb (hunk 3 above)
Comment (haus) [inline lib/vanagon/platform/rpm.rb]: For rpms, they can list a gpg key in the repo file and yum will automatically fetch it and ask on the first use of the repo.

Comment (melissa) [inline lib/vanagon/platform/rpm.rb]: Ah, I didn't realize that. Cool.

**Comment by haus** [comment]
will test on AIX this am

**Comment by haus** [comment]
AIX seems to be going fine so far. will know more once this build finishes.

**Comment by haus** [comment]
aix works.

**Comment by melissa** [comment]
:+1: wooo

**Comment by stahnma** [comment]
@haus rebase and I'll hit a button.

**Pull request merged**